bon-macros/src/builder/builder_gen/mod.rs

```diff
@@ -197,7 +197,7 @@ impl BuilderGenCtx {
 
     fn builder_impl(&self) -> Result<TokenStream2> {
         let finish_method = self.finish_method()?;
-        let (setter_methods, items_for_rustdoc) = self.setter_methods()?;
+        let (setget_methods, items_for_rustdoc) = self.setget_methods()?;
```

Comment:
I think the name `accessor` fits better here as the generalization of getters and setters and this name will also scale. For example, if we add a way to mutate the fields like in [`std::process::Command`](https://doc.rust-lang.org/stable/std/process/struct.Command.html#method.arg) builder where you can invoke `.arg()` multiple time and it pushes a new argument every time the method is called. In [`typed-builder`](https://docs.rs/typed-builder/latest/typed_builder/derive.TypedBuilder.html)'s terminology such methods are called mutators (search their docs page by the word `mutator`).

I'm also planning to have some way to define mutators in `bon`, I'd like to avoid the syntax of defining an `impl` block inside of the attribute that's used inside of `typed-builder`. A more nature way to express that API in that I'm thinking to have in the future is by letting the user write an `impl` block agains the builder type directly.

```suggestion
        let (accessor_methods, items_for_rustdoc) = self.accessor_methods()?;
```

Reply:
ok,

btw

```
    // Mutator has access to `x` additionally.
    #[mutator(requires = [x])]
    fn x_into_b(&mut self) {
        self.b.push(self.x)
    }
```

that is terrible, if i am going to rename `x`, will need to change macro? and x already is used in code, so seems they instead of doing some more automatic thing, went too way manual, so why to use builder, better write manual